substrate/client/src/client.rs

```diff
@@ -154,6 +154,24 @@ pub struct JustifiedHeader<Block: BlockT> {
 }
 
 impl<Block: BlockT> JustifiedHeader<Block> {
+	/// Creates new JustifiedHeader with given header, unchecked justification and authorities.
+	pub fn new(header: <Block as BlockT>::Header,
+			   justification: ::bft::UncheckedJustification<Block::Hash>,
+			   authorities: Vec<AuthorityId>,
+	) -> error::Result<Self> {
+		let parent_hash = header.parent_hash().clone();
+		let just = ::bft::check_justification::<Block>(&authorities[..], parent_hash, justification)
```

Comment:
Perhaps `just` can be named `justification` so line 169 becomes

```rust
Ok(JustifiedHeader {
  header,
  justification,
  authorities
})
```

Reply:
Sure, why not? :)

Reply:
done